pkg/kubelet/kubelet.go

```diff
@@ -2008,11 +2011,25 @@ func (kl *Kubelet) cleanupOrphanedVolumes(pods []*api.Pod, runningPods []*kubeco
 
 			// volume is unmounted.  some volumes also require detachment from the node.
 			if cleaner.Detacher != nil && len(refs) == 1 {
+
 				detacher := *cleaner.Detacher
-				err = detacher.Detach()
+				devicePath, _, err := mount.GetDeviceNameFromMount(kl.mounter, refs[0])
+				if err != nil {
+					glog.Errorf("Could not find device path %v", err)
+				}
+
+				if err = detacher.UnmountDevice(refs[0], kl.mounter); err != nil {
+					glog.Errorf("Could not unmount the global mount for %q: %v", name, err)
+				}
+
+				err = detacher.Detach(refs[0], kl.hostname)
 				if err != nil {
 					glog.Errorf("Could not detach volume %q: %v", name, err)
 				}
+
+				if err = detacher.WaitForDetach(devicePath, maxWaitForVolumeOps); err != nil {
```

Comment:
`WaitForDetach` should not be called here--because it blocks the sync loop. Since it is not actually implemented by any plugin yet, you could just put a comment here.